homeassistant/components/tradfri/cover.py

```diff
@@ -0,0 +1,142 @@
+"""Support for IKEA Tradfri covers."""
+import logging
+
+from pytradfri.error import PytradfriError
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION
+from homeassistant.core import callback
+from . import DOMAIN as TRADFRI_DOMAIN, KEY_API, KEY_GATEWAY
+from .const import CONF_GATEWAY_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Load Tradfri covers based on a config entry."""
+    gateway_id = config_entry.data[CONF_GATEWAY_ID]
+    api = hass.data[KEY_API][config_entry.entry_id]
+    gateway = hass.data[KEY_GATEWAY][config_entry.entry_id]
+
+    devices_commands = await api(gateway.get_devices())
+    devices = await api(devices_commands)
+    covers = [dev for dev in devices if dev.has_blind_control]
+    if covers:
+        async_add_entities(TradfriCover(cover, api, gateway_id) for cover in covers)
+
+
+class TradfriCover(CoverDevice):
+    """The platform class required by Home Assistant."""
+
+    def __init__(self, cover, api, gateway_id):
+        """Initialize a cover."""
+        self._api = api
+        self._unique_id = f"{gateway_id}-{cover.id}"
+        self._cover = None
+        self._cover_control = None
+        self._cover_data = None
+        self._name = None
+        self._available = True
+        self._gateway_id = gateway_id
+
+        self._refresh(cover)
+
+    @property
+    def unique_id(self):
+        """Return unique ID for cover."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        info = self._cover.device_info
+
+        return {
+            "identifiers": {(TRADFRI_DOMAIN, self._cover.id)},
+            "name": self._name,
+            "manufacturer": info.manufacturer,
+            "model": info.model_number,
+            "sw_version": info.firmware_version,
+            "via_device": (TRADFRI_DOMAIN, self._gateway_id),
+        }
+
+    async def async_added_to_hass(self):
+        """Start thread when added to hass."""
+        self._async_start_observe()
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def should_poll(self):
+        """No polling needed for tradfri cover."""
+        return False
+
+    @property
+    def name(self):
+        """Return the display name of this cover."""
+        return self._name
+
+    @property
+    def current_cover_position(self):
+        """Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return self._cover_data.current_cover_position
+
+    def set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        self._cover_control.set_state(kwargs[ATTR_POSITION])
```

Comment:
I think we need to use the api instance to handle the command instance that is returned when we call `_cover_control.set_state`.

Compare with the light platform:
https://github.com/home-assistant/home-assistant/blob/15bb12f48eddf2416519f6568da1f1b118aeb533/homeassistant/components/tradfri/light.py#L256

Reply:
Good catch - I've updated the method. Should I use `await` though? It looks like the whole method is async in the light-example?

@robbinonline can you please download the cover-file again and give it another go?

Reply:
Yes, we need to use `await`.

Reply:
And we need to update all the actuator methods.

Reply:
Does the last comment mean putting async in front of all methods?

Reply:
Make these methods async and rename them to `async_*`: `set_cover_position`
`open_cover`
`close_cover`

Reply:
Oh, I think I figured out what you meant